game/scripts/vscripts/modifiers/modifier_scan_true_sight.lua

```diff
@@ -0,0 +1,65 @@
+---
+--- Created by Zarnotox.
+--- DateTime: 29-Nov-17 10:50
+---
+
+modifier_scan_true_sight_thinker = class( ModifierBaseClass )
+modifier_scan_true_sight = class( ModifierBaseClass )
+
+--------- modifier_scan_true_sight_thinker ---------
+
+function modifier_scan_true_sight_thinker:IsPurgable()
+    return false
+end
+
+if IsServer() then
+
+    function modifier_scan_true_sight_thinker:OnCreated( event )
+
+        self.debuff_interval = 0.4
+        self.debuff_duration = 0.5
+        self.radius = SCAN_REVEAL_RADIUS
+
+        self:StartIntervalThink(self.debuff_interval)
```

Comment:
Though it is called a modifier thinker, you could just write this as an aura. Though I don't know if it's necessarily worth the effort now.

Reply:
I can still do that, have to look up how auras work first though